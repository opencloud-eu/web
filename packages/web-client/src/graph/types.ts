import type { AxiosInstance, ResponseType } from 'axios'
import type { Configuration } from './generated'

export interface GraphFactoryOptions {
  axiosClient: AxiosInstance
  config: Configuration
}

export interface GraphRequestOptions {
  headers?: Record<string, string>
  params?: Record<string, string>
  signal?: AbortSignal
  responseType?: ResponseType
}
