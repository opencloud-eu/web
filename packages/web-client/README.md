# web-client

The `web-client` is a standalone package that allows you to interact with the [OpenCloud](https://github.com/opencloud-eu/opencloud/) APIs via TypeScript. It provides an abstraction layer between the server and a (web-) application that converts API data into objects with helpful types and utilities. This abstraction ensures that users of the APIs don't need in-depth knowledge about them, such as required methods or returned status codes.

The supported APIs are:

- Graph (drives, sharing, user & group management)
- OCS (capabilities & url signing)
- WebDAV (file operations)

## Installation

Depending on your package manager, run one of the following commands:

```
$ npm install @opencloud-eu/web-client

$ pnpm add @opencloud-eu/web-client

$ yarn add @opencloud-eu/web-client
```

## Usage

### Graph

The graph client needs to be instantiated with a base URI corresponding to your OpenCloud deployment and an axios instance. The axios instance is being used for all requests, which means it needs to include all relevant headers either statically or via interceptor.

```
import axios from axios
import { graph } from '@opencloud-eu/web-client'

const accessToken = 'some_access_token'
const baseURI = 'some_base_uri'

const axiosClient = axios.create({
	headers: { Authorization: accessToken }
})

const graphClient = graph(baseURI, axiosClient)
```

The following example demonstrates how to retrieve all spaces accessible to the user. A `SpaceResource` can then be used to e.g. fetch files and folders (see webdav example down below).

```
const mySpaces = await graphClient.drives.listMyDrives()
```

### OCS

The ocs client needs to be instantiated with a base URI corresponding to your OpenCloud deployment and an axios instance. The axios instance is being used for all requests, which means it needs to include all relevant headers either statically or via interceptor.

```
import axios from axios
import { ocs } from '@opencloud-eu/web-client'

const accessToken = 'some_access_token'
const baseURI = 'some_base_uri'

const axiosClient = axios.create({
	headers: { Authorization: accessToken }
})

const ocsClient = ocs(baseURI, axiosClient)
```

The following examples demonstrate how to fetch capabilities and sign URLs.

```
const capabilities = await ocsClient.getCapabilities()

const signedUrl = await ocsClient.signUrl('some_url_to_sign', 'your_username')
```

### WebDav

The webdav client needs to be instantiated with a base URI corresponding to your OpenCloud deployment. You can also pass a header callback which will be called with every dav request.

```
import { webdav } from '@opencloud-eu/web-client'

const accessToken = 'some_access_token'
const baseURI = 'some_base_uri'

const webDavClient = webdav(
	baseURI,
	() => ({ Authorization: accessToken })
)
```

The following example demonstrates how to list all resources of a given `SpaceResource` (see above how to fetch space resources).

```
const { resource, children } = await webDavClient.listFiles(spaceResource)
```
