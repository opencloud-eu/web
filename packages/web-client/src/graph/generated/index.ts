/* tslint:disable */
/* eslint-disable */
/**
 * Libre Graph API
 * Libre Graph is a free API for cloud collaboration inspired by the MS Graph API.
 *
 * The version of the OpenAPI document: v1.0.7
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


export * from "./api";
export * from "./configuration";

