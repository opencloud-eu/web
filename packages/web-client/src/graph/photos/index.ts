export * from './photo'
export * from './types'
