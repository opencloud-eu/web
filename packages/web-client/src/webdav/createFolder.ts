import { FolderResource, SpaceResource } from '../helpers'
import { GetFileInfoFactory } from './getFileInfo'
import { DAV, DAVRequestOptions } from './client/dav'
import { WebDavOptions } from './types'
import { getWebDavPath } from './utils'

type CreateFolderOptions = {
  path: string
  parentFolderId?: string
  folderName?: string
  fetchFolder?: boolean
} & DAVRequestOptions

export const CreateFolderFactory = (
  dav: DAV,
  getFileInfoFactory: ReturnType<typeof GetFileInfoFactory>,
  options: WebDavOptions
) => {
  return {
    async createFolder(
      space: SpaceResource,
      { path, parentFolderId, folderName, fetchFolder = true, ...opts }: CreateFolderOptions
    ): Promise<FolderResource> {
      const webDavPath = getWebDavPath(space, { fileId: parentFolderId, path, name: folderName })
      await dav.mkcol(webDavPath)

      if (fetchFolder) {
        // FIXME: mkcol doesn't return a fileId on server side
        return getFileInfoFactory.getFileInfo(space, { path }, opts)
      }
    }
  }
}
