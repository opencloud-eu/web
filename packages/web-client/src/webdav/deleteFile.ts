import { urlJoin } from '../utils'
import { SpaceResource } from '../helpers'
import { WebDavOptions } from './types'
import { DAV, DAVRequestOptions } from './client'

export const DeleteFileFactory = (dav: DAV, options: WebDavOptions) => {
  return {
    deleteFile(space: SpaceResource, { path, ...opts }: { path: string } & DAVRequestOptions) {
      // FIXME: use id-based
      return dav.delete(urlJoin(space.webDavPath, path), opts)
    }
  }
}
