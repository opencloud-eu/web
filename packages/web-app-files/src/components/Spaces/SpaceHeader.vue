<template>
  <div
    class="space-header oc-p-m"
    :class="{ 'oc-flex': !imageExpanded && !isMobileWidth, 'space-header-squashed': isSideBarOpen }"
  >
    <div
      class="space-header-image"
      :class="{ 'space-header-image-expanded': imageExpanded || isMobileWidth }"
    >
      <div
        v-if="imagesLoading.includes(space.id)"
        class="oc-height-1-1 oc-flex oc-flex-middle oc-flex-center"
      >
        <oc-spinner :aria-label="$gettext('Space image is loading')" />
      </div>
      <img
        v-else-if="imageContent"
        class="oc-cursor-pointer"
        alt=""
        :src="imageContent"
        @click="toggleImageExpanded"
      />
      <div v-else class="space-header-image-default oc-flex oc-flex-middle oc-flex-center">
        <oc-icon name="layout-grid" size="xxlarge" class="oc-px-m oc-py-m" />
      </div>
    </div>
    <div class="space-header-infos">
      <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
        <div class="oc-flex oc-flex-middle space-header-infos-heading">
          <h2 class="space-header-name">{{ space.name }}</h2>
          <oc-button
            :id="`space-context-btn`"
            v-oc-tooltip="$gettext('Show context menu')"
            :aria-label="$gettext('Show context menu')"
            appearance="raw"
            class="oc-ml-s oc-p-xs"
          >
            <oc-icon name="more-2" />
          </oc-button>
          <oc-drop
            :drop-id="`space-context-drop`"
            :toggle="`#space-context-btn`"
            mode="click"
            close-on-click
            :options="{ delayHide: 0 }"
            padding-size="small"
            position="right-start"
          >
            <space-context-actions :action-options="{ resources: [space] }" />
          </oc-drop>
        </div>
        <oc-button
          v-if="memberCount"
          :aria-label="$gettext('Open context menu and show members')"
          appearance="raw"
          no-hover
          @click="openSideBarSharePanel"
        >
          <oc-icon name="group" fill-type="line" size="small" />
          <span
            v-if="memberCount"
            class="space-header-people-count oc-text-small"
            v-text="memberCountString"
          />
          <oc-spinner v-else size="small" :aria-label="$gettext('Loading members')" />
        </oc-button>
      </div>
      <p v-if="space.description" class="oc-mt-rm oc-text-bold">{{ space.description }}</p>
      <div
        v-if="readmesLoading.includes(space.id)"
        class="space-header-readme-loading oc-flex oc-flex-middle oc-flex-center"
      >
        <oc-spinner :aria-label="$gettext('Space description is loading')" />
      </div>
      <div
        v-else-if="markdownResource && markdownContent"
        ref="markdownContainerRef"
        class="markdown-container oc-flex"
      >
        <text-editor
          class="markdown-container-content"
          is-read-only
          :current-content="markdownContent"
        />
        <div class="markdown-container-edit oc-ml-s">
          <oc-button
            type="router-link"
            size="small"
            :aria-label="$gettext('Edit description')"
            appearance="raw"
            class="oc-p-xs"
            :to="editReadMeContentLink"
          >
            <oc-icon name="pencil" size="small" fill-type="line" />
          </oc-button>
        </div>
      </div>
      <div
        v-if="showMarkdownCollapse && markdownContent"
        class="markdown-collapse oc-text-center oc-mt-s"
      >
        <oc-button appearance="raw" no-hover @click="toggleMarkdownCollapsed">
          <span>{{ toggleMarkdownCollapsedText }}</span>
        </oc-button>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed, inject, nextTick, onBeforeUnmount, onMounted, Ref, ref, unref, watch } from 'vue'
import { buildSpaceImageResource, Resource, SpaceResource } from '@opencloud-eu/web-client'
import {
  useClientService,
  ProcessorType,
  useResourcesStore,
  TextEditor,
  useFileActions,
  useLoadPreview,
  useSpacesStore,
  useSharesStore
} from '@opencloud-eu/web-pkg'
import { ImageDimension } from '@opencloud-eu/web-pkg'
import SpaceContextActions from './SpaceContextActions.vue'
import { eventBus } from '@opencloud-eu/web-pkg'
import { SideBarEventTopics } from '@opencloud-eu/web-pkg'
import { useGettext } from 'vue3-gettext'
import { DriveItem } from '@opencloud-eu/web-client/graph/generated'
import { storeToRefs } from 'pinia'

const markdownContainerCollapsedClass = 'collapsed'

const { space, isSideBarOpen = false } = defineProps<{
  space: SpaceResource
  isSideBarOpen?: boolean
}>()

const language = useGettext()
const { $gettext, $ngettext } = language
const clientService = useClientService()
const { getFileContents, getFileInfo } = clientService.webdav
const resourcesStore = useResourcesStore()
const { getDefaultAction } = useFileActions()
const { loadPreview } = useLoadPreview()
const spacesStore = useSpacesStore()
const sharesStore = useSharesStore()
const { imagesLoading, readmesLoading } = storeToRefs(spacesStore)

const isMobileWidth = inject<Ref<boolean>>('isMobileWidth')

const markdownContainerRef = ref(null)
const markdownContent = ref('')
const markdownResource = ref<Resource>(null)
const markdownCollapsed = ref(true)
const showMarkdownCollapse = ref(false)
const toggleMarkdownCollapsedText = computed(() => {
  return unref(markdownCollapsed) ? $gettext('Show more') : $gettext('Show less')
})
const toggleMarkdownCollapsed = () => {
  markdownCollapsed.value = !unref(markdownCollapsed)
  unref(markdownContainerRef).classList.toggle(markdownContainerCollapsedClass)
}
const onMarkdownResize = () => {
  if (!unref(markdownContainerRef)) {
    return
  }

  unref(markdownContainerRef).classList.remove(markdownContainerCollapsedClass)
  const markdownContainerHeight = unref(markdownContainerRef).offsetHeight
  if (markdownContainerHeight < 150) {
    showMarkdownCollapse.value = false
    return
  }
  showMarkdownCollapse.value = true

  if (unref(markdownCollapsed)) {
    unref(markdownContainerRef).classList.add(markdownContainerCollapsedClass)
  }
}
const markdownResizeObserver = new ResizeObserver(onMarkdownResize)
const observeMarkdownContainerResize = () => {
  if (!markdownResizeObserver || !unref(markdownContainerRef)) {
    return
  }
  markdownResizeObserver.unobserve(unref(markdownContainerRef))
  markdownResizeObserver.observe(unref(markdownContainerRef))
}
const unobserveMarkdownContainerResize = () => {
  if (!markdownResizeObserver || !unref(markdownContainerRef)) {
    return
  }
  markdownResizeObserver.unobserve(unref(markdownContainerRef))
}

const memberCount = ref<number>()
watch(
  () => sharesStore.collaboratorShares.length,
  async () => {
    // set space member count
    try {
      const { count } = await clientService.graphAuthenticated.permissions.listPermissions(
        space.id,
        space.id,
        sharesStore.graphRoles,
        { count: true, filter: "grantedToV2 ne ''" }
      )
      memberCount.value = count || 1
    } catch (e) {
      console.error(e)
    }
  },
  { immediate: true }
)

onMounted(observeMarkdownContainerResize)
onBeforeUnmount(() => {
  unobserveMarkdownContainerResize()
  spacesStore.purgeReadmesLoading()
})

const loadReadmeContent = async () => {
  spacesStore.addToReadmesLoading(space.id)

  try {
    const fileContentsResponse = await getFileContents(space, {
      path: `.space/${space.spaceReadmeData.name}`
    })

    const fileInfoResponse = await getFileInfo(space, {
      path: `.space/${space.spaceReadmeData.name}`
    })

    unobserveMarkdownContainerResize()
    markdownContent.value = fileContentsResponse.body
    markdownResource.value = fileInfoResponse
    spacesStore.removeFromReadmesLoading(space.id)

    await nextTick()
    if (unref(markdownContent)) {
      observeMarkdownContainerResize()
    }
  } catch (e) {
    if ([425, 429].includes(e.statusCode)) {
      const retryAfter = e.response?.headers?.['retry-after'] || 5
      await new Promise((resolve) => setTimeout(resolve, retryAfter * 1000))
      return loadReadmeContent()
    }

    spacesStore.removeFromReadmesLoading(space.id)
    console.error(e)
  }
}

watch(
  computed(() => space.spaceReadmeData),
  async (data: DriveItem) => {
    if (!data) {
      return
    }

    await loadReadmeContent()
  },
  { deep: true, immediate: true }
)

const imageContent = ref<string>(null)
const imageExpanded = ref(false)

const editReadMeContentLink = computed(() => {
  const action = getDefaultAction({ resources: [unref(markdownResource)], space })

  if (!action.route) {
    return null
  }

  return action.route({ space, resources: [unref(markdownResource)] })
})
const toggleImageExpanded = () => {
  imageExpanded.value = !unref(imageExpanded)
}

watch(
  computed(() => space.spaceImageData),
  async (data) => {
    if (!data) {
      return
    }
    const resource = buildSpaceImageResource(space)
    imageContent.value = await loadPreview({
      space,
      resource,
      dimensions: ImageDimension.Tile,
      processor: ProcessorType.enum.fit,
      cancelRunning: true,
      updateStore: false
    })
  },
  { immediate: true }
)

const memberCountString = computed(() => {
  return $ngettext('%{count} member', '%{count} members', unref(memberCount), {
    count: unref(memberCount).toString()
  })
})

const openSideBarSharePanel = () => {
  resourcesStore.setSelection([])
  eventBus.publish(SideBarEventTopics.openWithPanel, 'space-share')
}
</script>

<style lang="scss">
.space-header {
  &-squashed {
    .space-header-image {
      @media only screen and (max-width: 1200px) {
        display: none;
      }
    }
  }

  &-image {
    width: 280px;
    min-width: 280px;
    aspect-ratio: 16 / 9;
    margin-right: var(--oc-space-large);
    max-height: 158px;

    &-default {
      height: 100%;
      border-radius: 10px;
    }

    &-expanded {
      width: 100%;
      margin: 0;
      max-height: 100%;
      max-width: 100%;
    }

    img {
      border-radius: 10px;
      height: 100%;
      width: 100%;
      max-height: 100%;
      object-fit: cover;
    }
  }

  &-infos {
    flex: 1;

    &-heading {
      max-width: 100%;
    }
  }

  &-name {
    font-size: 1.5rem;
    word-break: break-all;
  }

  &-people-count {
    white-space: nowrap;
  }

  .markdown-container {
    &-content {
      .md-editor-preview-wrapper {
        padding: 0;
      }
    }
  }

  .markdown-container.collapsed {
    max-height: 100px;
    overflow: hidden;
    -webkit-mask-image: linear-gradient(180deg, #000 90%, transparent);
  }
}
</style>
