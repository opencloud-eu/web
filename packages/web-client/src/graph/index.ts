import { AxiosInstance } from 'axios'
import { Configuration } from './generated'
import { type GraphUsers, UsersFactory } from './users'
import { type GraphGroups, GroupsFactory } from './groups'
import { ApplicationsFactory, GraphApplications } from './applications'
import { DrivesFactory, GraphDrives } from './drives'
import { DriveItemsFactory, GraphDriveItems } from './driveItems'
import { TagsFactory, GraphTags } from './tags'
import { ActivitiesFactory, GraphActivities } from './activities'
import { PermissionsFactory, GraphPermissions } from './permissions'
import { PhotoFactory, GraphPhotos } from './photos'

export interface Graph {
  activities: GraphActivities
  applications: GraphApplications
  tags: GraphTags
  drives: GraphDrives
  driveItems: GraphDriveItems
  users: GraphUsers
  groups: GraphGroups
  permissions: GraphPermissions
  photos: GraphPhotos
}

export const graph = (baseURI: string, axiosClient: AxiosInstance): Graph => {
  const url = new URL(baseURI)
  url.pathname = [...url.pathname.split('/'), 'graph'].filter(Boolean).join('/')
  const config = new Configuration({
    basePath: url.href
  })

  return <Graph>{
    activities: ActivitiesFactory({ axiosClient, config }),
    applications: ApplicationsFactory({ axiosClient, config }),
    tags: TagsFactory({ axiosClient, config }),
    drives: DrivesFactory({ axiosClient, config }),
    driveItems: DriveItemsFactory({ axiosClient, config }),
    users: UsersFactory({ axiosClient, config }),
    groups: GroupsFactory({ axiosClient, config }),
    permissions: PermissionsFactory({ axiosClient, config }),
    photos: PhotoFactory({ axiosClient, config })
  }
}
