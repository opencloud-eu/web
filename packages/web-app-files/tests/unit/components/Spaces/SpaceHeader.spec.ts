import { ref } from 'vue'
import SpaceHeader from '../../../../src/components/Spaces/SpaceHeader.vue'
import { DriveItem } from '@opencloud-eu/web-client/graph/generated'
import { SpaceResource, Resource, buildSpaceImageResource } from '@opencloud-eu/web-client'
import { defaultPlugins, mount, defaultComponentMocks } from '@opencloud-eu/web-test-helpers'
import { mock } from 'vitest-mock-extended'
import { GetFileContentsResponse } from '@opencloud-eu/web-client/webdav'
import { flushPromises } from '@vue/test-utils'

vi.mock('@opencloud-eu/web-pkg', async (importOriginal) => ({
  ...(await importOriginal<any>()),
  useFileActions: vi.fn().mockReturnValue({
    getDefaultAction: vi.fn().mockReturnValue({ handler: vi.fn() })
  }),
  useLoadPreview: vi.fn().mockReturnValue({
    loadPreview: vi.fn(() => 'blob:image')
  })
}))

vi.mock('@opencloud-eu/web-client', async (importOriginal) => ({
  ...(await importOriginal<any>()),
  buildSpaceImageResource: vi.fn()
}))

const getSpaceMock = (spaceImageData: DriveItem = undefined) =>
  mock<SpaceResource>({
    id: '1',
    name: 'space',
    description: '',
    spaceReadmeData: undefined,
    spaceImageData
  })

describe('SpaceHeader', () => {
  it('should add the "squashed"-class when the sidebar is opened', async () => {
    const wrapper = getWrapper({ space: getSpaceMock(), isSideBarOpen: true })
    await flushPromises()
    expect(wrapper.find('.space-header-squashed').exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
  describe('space image', () => {
    it('should show the set image', async () => {
      const wrapper = getWrapper({ space: getSpaceMock({ webDavUrl: '/' }) })
      await flushPromises()
      expect(wrapper.find('.space-header-image img').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('should take full width in mobile view', () => {
      const wrapper = getWrapper({
        space: getSpaceMock({ webDavUrl: '/' }),
        isMobileWidth: true
      })
      expect(wrapper.find('.space-header').attributes('class')).not.toContain('oc-flex')
      expect(wrapper.find('.space-header-image').attributes('class')).toContain(
        'space-header-image-expanded'
      )
    })
    it('shows a loading spinner when the image is loading', () => {
      const space = getSpaceMock()
      const wrapper = getWrapper({ space, imagesLoading: [space.id] })
      expect(wrapper.find('.space-header-image .oc-spinner').exists()).toBeTruthy()
    })
  })
  describe('space description', () => {
    it('should show the description', async () => {
      const space = getSpaceMock()
      space.spaceReadmeData = {}
      const wrapper = getWrapper({ space })
      await flushPromises()
      expect(wrapper.find('.markdown-container').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('shows a loading spinner when the description is loading', () => {
      const space = getSpaceMock()
      space.spaceReadmeData = {}
      const wrapper = getWrapper({ space, readmesLoading: [space.id] })
      expect(wrapper.find('.space-header-readme-loading').exists()).toBeTruthy()
    })
  })
  describe('space member count', () => {
    it('should show the correct amount of space members', async () => {
      const space = getSpaceMock()
      space.spaceReadmeData = {}
      const wrapper = getWrapper({ space, memberCount: 5 })
      await flushPromises()
      expect(wrapper.find('.space-header-people-count').text()).toContain('5')
    })
  })
})

function getWrapper({
  space = {} as SpaceResource,
  isSideBarOpen = false,
  isMobileWidth = false,
  imagesLoading = [],
  readmesLoading = [],
  memberCount = 0
}) {
  const mocks = defaultComponentMocks()
  mocks.$previewService.loadPreview.mockResolvedValue('blob:image')
  vi.mocked(buildSpaceImageResource).mockReturnValue(mock<Resource>())

  mocks.$clientService.webdav.getFileContents.mockResolvedValue(
    mock<GetFileContentsResponse>({ body: 'body' })
  )
  mocks.$clientService.webdav.getFileInfo.mockResolvedValue(mock<Resource>())
  mocks.$clientService.graphAuthenticated.permissions.listPermissions.mockResolvedValue({
    shares: [],
    allowedActions: [],
    allowedRoles: [],
    count: memberCount
  })

  return mount(SpaceHeader, {
    props: {
      space,
      isSideBarOpen
    },
    global: {
      mocks,
      plugins: [
        ...defaultPlugins({ piniaOptions: { spacesState: { imagesLoading, readmesLoading } } })
      ],
      provide: { ...mocks, isMobileWidth: ref(isMobileWidth) },
      stubs: {
        'space-context-actions': true
      }
    }
  })
}
