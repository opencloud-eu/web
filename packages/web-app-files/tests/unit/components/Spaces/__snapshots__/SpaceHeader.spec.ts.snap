// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpaceHeader > should add the "squashed"-class when the sidebar is opened 1`] = `
"<div class="space-header oc-p-m oc-flex space-header-squashed">
  <div class="space-header-image"><img class="oc-cursor-pointer" alt="" src="blob:image"></div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name">space</h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-ml-s oc-p-xs" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m"><svg data-testid="inline-svg-stub" src="icons/more-2-fill.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]" loading="false"></space-context-actions-stub>
          </div>
        </div>
      </div> <button type="button" aria-label="Open context menu and show members" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw no-hover">
        <!--v-if-->
        <!-- @slot Content of the button --> <span class="oc-icon oc-icon-s"><svg data-testid="inline-svg-stub" src="icons/group-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span> <span class="space-header-people-count oc-text-small">1 member</span>
      </button>
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`SpaceHeader > space description > should show the description 1`] = `
"<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image"><img class="oc-cursor-pointer" alt="" src="blob:image"></div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name">space</h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-ml-s oc-p-xs" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m"><svg data-testid="inline-svg-stub" src="icons/more-2-fill.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]" loading="false"></space-context-actions-stub>
          </div>
        </div>
      </div> <button type="button" aria-label="Open context menu and show members" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw no-hover">
        <!--v-if-->
        <!-- @slot Content of the button --> <span class="oc-icon oc-icon-s"><svg data-testid="inline-svg-stub" src="icons/group-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span> <span class="space-header-people-count oc-text-small">1 member</span>
      </button>
    </div>
    <!--v-if-->
    <div class="markdown-container oc-flex">
      <!---->
      <div class="markdown-container-edit oc-ml-s"><a attrs="[object Object]" aria-label="Edit description" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-p-xs"></a></div>
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`SpaceHeader > space image > should show the set image 1`] = `
"<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image"><img class="oc-cursor-pointer" alt="" src="blob:image"></div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name">space</h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-ml-s oc-p-xs" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m"><svg data-testid="inline-svg-stub" src="icons/more-2-fill.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]" loading="false"></space-context-actions-stub>
          </div>
        </div>
      </div> <button type="button" aria-label="Open context menu and show members" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw no-hover">
        <!--v-if-->
        <!-- @slot Content of the button --> <span class="oc-icon oc-icon-s"><svg data-testid="inline-svg-stub" src="icons/group-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span> <span class="space-header-people-count oc-text-small">1 member</span>
      </button>
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
